panoply/error_normalization.py

```diff
@@ -0,0 +1,97 @@
+from panoply import DataSource, PanoplyException
+from datetime import datetime
+
+
+ERROR_CODES_REGISTRY = {
+    400: 'Bad request',
+    401: 'Unauthorized',
+    403: 'Permissions error',
+    404: 'Resource is not found',
+    408: 'Timeout',
+    422: 'Parsing error',
+    429: 'Rate limit',
+    500: 'Internal exception',
+    501: 'Unclassified error',
+    506: 'SDK error in processing an error'
+}
+
+EXCEPTIONS_REGISTRY = {}
+
+
+class NormalizedException(PanoplyException):
+    def __init__(self, message, code, exception_cls,
+                 phase, source_type, source_id, database_id,
+                 retryable=True):
+        super().__init__(message, retryable)
+        self.message = message
+        self.code = code
+        self.phase = phase
```

Comment:
How is `phase` used? It seems to be just an arbitrary string.

Reply:
Will replace it with some relevant structure.

Reply:
Ok... structured or not I still don't know what purpose phase is serving 😃

Reply:
In the [`handle_errors`](https://github.com/panoplyio/panoply-python-sdk/blob/30f742e150a5a95166cb5457d8aacf57ef16b531/panoply/error_normalization.py#L59) there is a description of this param. It is set to distinguish between collect and configuration stages.

Reply:
Got it. So because you're setting it manually per function it seems to me that it'll require developers to create separate functions for config and collect parts, is this the intention?

Reply:
Yes, we want to separate configuration utils for UI and collect process.